catroid/src/main/java/org/catrobat/catroid/content/bricks/ShowTextColorSizeAlignmentBrick.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.bricks;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.view.View;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.common.Nameable;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.actions.ScriptSequenceAction;
+import org.catrobat.catroid.content.bricks.brickspinner.BrickSpinner;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShowTextColorSizeAlignmentBrick extends UserVariableBrick {
+	private static final long serialVersionUID = 1L;
+	public static final int ALIGNMENT_STYLE_LEFT = 0;
+	public static final int ALIGNMENT_STYLE_CENTERED = 1;
+	public static final int ALIGNMENT_STYLE_RIGHT = 2;
+	public int alignmentSelection = ALIGNMENT_STYLE_CENTERED;
+
+	class AlignmentStyle implements Nameable, BrickSpinner.OnItemSelectedListener<AlignmentStyle> {
```

Comment:
Actually the `ShowTextColorSizeAlignmentBrick` should implement `BrickSpinner.OnItemSelectedListener<AlignmentStyle>` so that the brick itself can react to spinner selection changes. then you can also remove the reference to the brick (`ShowTextColorSizeAlignmentBrick object;`) from the subclass. and besides if you want to access a member variable of the parent class from a subclass you should just use `ShowTextColorSizeAlignmentBrick.this` instead.

Reply:
Your suggestion was my first approach but cannot `ShowTextColorSizeAlignmentBrick` cannot implement BrickSpinner.OnItemSelectedListener<AlignmentStyle> because it extents UserVariable which implements BrickSpinner.OnItemSelectedListener.  But thx for the second advice :)

Reply:
Oh yes. I missed that it there are going to be two different spinners in this brick. still it does not make sense to have the objects that are part of the spinner and therefore in its list implement the listener.  I suggest you solve it by adding another class that implements the listener to the brick. 

```
        ...
	public static final int ALIGNMENT_STYLE_CENTERED = 1;
	public static final int ALIGNMENT_STYLE_RIGHT = 2;
	public int alignmentSelection = ALIGNMENT_STYLE_CENTERED;     
  
        private transient SpinnerListener spinnerListener = new SpinnerListener();
	
	class SpinnerListener implements BrickSpinner.OnItemSelectedListener<AlignmentStyle> {
		
		@Override
		public void onStringOptionSelected(String string) {
		}

		@Override
		public void onItemSelected(@Nullable AlignmentStyle item) {
			ShowTextColorSizeAlignmentBrick.this.alignmentSelection = item.alignmentStyle;
		}

		@Override
		public void onNewOptionSelected() {
		}
	}
	class AlignmentStyle implements Nameable, BrickSpinner.OnItemSelectedListener<AlignmentStyle> {

		AlignmentStyle(String name, int alignmentStyle) {
			this.name = name;
			this.alignmentStyle = alignmentStyle;
		}
        ...
```

Reply:
**Do not forget** to then override the `clone()` method and set the member that holds the `spinnerListener` to null, otherwise every newly cloned brick that is created from this one has the same listener and consequently gets called whenever something is changed in the original brick.

```
	@Override
	public BrickBaseType clone() throws CloneNotSupportedException {
		ShowTextColorSizeAlignmentBrick clone = (ShowTextColorSizeAlignmentBrick) super.clone();
		clone.spinnerListener = null;
		return clone;
	}

```

Reply:
Alternatively you could try and don't add a member variable for the spinner at all. I'm not sure if this works, because you can never be sure about what the garbage collector does in java. but perhaps you just have to add the class and then just do:

```
...
spinner.setOnItemSelectedListener(new SpinnerListener);
...
```
in `getView()`. or even move the whole class definition here, like this:

```
	@Override
	public View getView(Context context) {
		super.getView(context);
		List<Nameable> items = new ArrayList<>();
		items.add(new AlignmentStyle(context.getString(R.string.brick_show_variable_aligned_left),
				ALIGNMENT_STYLE_LEFT));
		items.add(new AlignmentStyle(context.getString(R.string.brick_show_variable_aligned_centered),
				ALIGNMENT_STYLE_CENTERED));
		items.add(new AlignmentStyle(context.getString(R.string.brick_show_variable_aligned_right),
				ALIGNMENT_STYLE_RIGHT));
		BrickSpinner<AlignmentStyle> spinner =
				new BrickSpinner<>(R.id.brick_show_variable_color_size_align_spinner, view, items);
		spinner.setSelection(alignmentSelection);

		spinner.setOnItemSelectedListener(new BrickSpinner.OnItemSelectedListener<AlignmentStyle> (){

			@Override
			public void onStringOptionSelected(String string) {
			}

			@Override
			public void onItemSelected(@Nullable AlignmentStyle item) {
				ShowTextColorSizeAlignmentBrick.this.alignmentSelection = item.alignmentStyle;
			}

			@Override
			public void onNewOptionSelected() {
			}
		});
		return view;
	}
```

so many options :confused: but now that I think about it again the last two might be the best -> here you wouldn't even have to override the `clone(..)` method.